traits-stubs/traits-stubs/trait_types.pyi

```diff
@@ -0,0 +1,426 @@
+from typing import (
+    Any as _Any,
+    Callable as _CallableType,
+    Dict as _Dict,
+    List as _List,
+    Optional,
+    Sequence as _Sequence,
+    Set as _Set,
+    Type as _Type,
+    TypeVar,
+    Union as _Union,
+)
+from uuid import UUID as _UUIDType
+
+from .trait_type import TraitType as TraitType
+from .trait_type import _TraitType
+
+MutableTypes: _Any
+SetTypes: _Any
+int_fast_validate: _Any
+float_fast_validate: _Any
+complex_fast_validate: _Any
+bool_fast_validate: _Any
+
+
+def default_text_editor(trait: _Any, type: Optional[_Any] = ...):
+    ...
+
+
+_T = TypeVar("_T")
+_S = TypeVar("_S")
+
+_Trait = _Union[_TraitType[_S, _T], _Type[_TraitType[_S, _T]]]
+
+
+class Any(TraitType):
+    ...
+
+
+# -----------------Int--------------------
+class _BaseInt(_TraitType[_T, int]):
+    ...
+
+
+class BaseInt(_BaseInt[int]):
+    default_value: int = ...
+
+
+class Int(BaseInt):
+    default_value: int = ...
+
+
+# -----------------Float--------------------
+class _BaseFloat(_TraitType[_T, float]):
+    ...
+
+
+class BaseFloat(_BaseFloat[float]):
+    default_value: float = ...
+
+
+class Float(BaseFloat):
+    default_value: float = ...
+
+
+# -----------------Complex--------------------
+class _BaseComplex(_TraitType[_T, complex]):
+    ...
+
+
+class BaseComplex(_BaseComplex[complex]):
+    default_value: complex = ...
+
+
+class Complex(BaseComplex):
+    default_value: complex = ...
+
+
+# -----------------Str--------------------
+
+class _BaseStr(_TraitType[_T, str]):
+    default_value: str = ...
+
+
+class BaseStr(_BaseStr[str]):
+    default_value: str = ...
+
+
+class Str(BaseStr):
+    default_value: str = ...
+
+
+# -----------------Title--------------------
+
+class Title(Str):
+    default_value: str = ...
+
+
+# -----------------Bytes--------------------
+class _BaseBytes(_TraitType[_T, bytes]):
+    ...
+
+
+class BaseBytes(_BaseBytes[bytes]):
+    default_value: bytes = ...
+
+
+class Bytes(BaseBytes):
+    default_value: bytes = ...
+
+
+# -----------------Bool--------------------
+class _BaseBool(_TraitType[_T, bool]):
+    ...
+
+
+class BaseBool(_BaseBool[bool]):
+    default_value: bool = ...
+
+
+class Bool(BaseBool):
+    default_value: bool = ...
+
+
+# -----------------BaseCInt--------------------
+
+class BaseCInt(BaseInt):
```

Comment:
"C" traits can accept a wider collection of types, so this should be `BaseCInt(_BaseInt[_Any])` at a first cut.  If you really want to get technical, it is anything that a single-argument `int()` call accepts, which according to TypeShed is `Union[Text, bytes, SupportsInt, _SupportsIndex]`:

https://github.com/python/typeshed/blob/91deb7a79f9eff58a8ff02566e7da9eb06a99ec6/stdlib/2and3/builtins.pyi#L140

where `_SupportsIndex` is here:
 https://github.com/python/typeshed/blob/91deb7a79f9eff58a8ff02566e7da9eb06a99ec6/stdlib/2and3/builtins.pyi#L37

(Actually, if you really want to get technical, `_SupportsIndex` is the `_Accepts` type of `BaseInt`)

Similar comments for other "C" traits below.  Let's do `_Any` for a first cut, and then we can refine.